Avoid jinja2 in when warnings

major opened the pull request:
This patch adjusts the handler to use the jinja2 statement directly
rather than using a variable with delimiters. This will hopefully
avoid the warning we currently see in OpenStack-Ansible:

    [WARNING]: when statements should not include jinja2 templating
    delimiters such as {{ }} or {% %}.
    Found: {{ groups['haproxy'] | default([]) | length > 0 }}

logan2211 commented on defaults/main.yml:
```diff
@@ -13,9 +14,6 @@
 # limitations under the License.
 
-# Can be used to completely disable the role's handler actions
-haproxy_management: "{{ haproxy_hosts | default([]) | length > 0 }}"
```
Let's leave this in and default it to yes so there is still a way for someone to completely disable the role actions even if haproxy is present.

logan2211 replied:
Actually I wonder if it might work to simply remove the jinja brackets from this var default and set it to:
```
haproxy_management: 'haproxy_hosts | default([]) | length > 0'
```

with no other changes

major replied:
You'd still need that jinja2 to be evaluated, though.

logan2211 commented on handlers/main.yml:
```diff
@@ -26,5 +27,5 @@
   delegate_to: "{{ item }}"
   with_items: "{{ haproxy_hosts }}"
-  when: haproxy_management | bool
+  when: haproxy_hosts | default([]) | length > 0
```
following on the above comment id suggest doing this:

```
when:
  - haproxy_hosts | default([]) | length > 0
  - haproxy_management | bool

logan2211 reviewed (changes requested):
Good catch.. Small suggestion to maintain the override ability to null out the handler if desired.

major committed:
Avoid jinja2 in when warnings

This patch adjusts the handler to use the jinja2 statement directly
rather than using a variable with delimiters. This will hopefully
avoid the warning we currently see in OpenStack-Ansible:

    [WARNING]: when statements should not include jinja2 templating
    delimiters such as {{ }} or {% %}.
    Found: {{ groups['haproxy'] | default([]) | length > 0 }}

major commented:
I tried another strategy. We do a check to see if there are haproxy hosts and we also check to see if `haproxy_management` has been set to `no`. Does that make sense?

logan2211 commented on defaults/main.yml:
```diff
@@ -14,7 +15,7 @@
 
 # Can be used to completely disable the role's handler actions
-haproxy_management: "{{ haproxy_hosts | default([]) | length > 0 }}"
+haproxy_management: yes 
```
Can you trim the whitespace at the end of the line

logan2211 reviewed (changes requested):
Yep LGTM that was my first suggestion exactly :D

Only one small nit..
